Microsoft.Toolkit.Win32/Microsoft.Toolkit.Forms.UI.XamlHost/WindowsXamlHostBase.cs

```diff
@@ -0,0 +1,182 @@
+        [Browsable(true)]
+        [Category("UWP XAML")]
+        [DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
+        [Description("Fired when UWP XAML content has been updated")]
+        public event EventHandler ChildChanged;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WindowsXamlHostBase"/> class.
+        /// </summary>
+        [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
+        public WindowsXamlHostBase()
+        {
+            SetStyle(ControlStyles.ContainerControl, true);
+            SetStyle(ControlStyles.SupportsTransparentBackColor, true);
+            SetStyle(ControlStyles.UserPaint, true);
+            SetStyle(ControlStyles.AllPaintingInWmPaint, true);
+
+            // Must be a container control with TabStop == false to allow nested UWP XAML Focus
+            // BUGBUG: Uncomment when nested Focus is available
+            // TabStop = false;
+
+            // Respond to size changes on this Control
+            SizeChanged += OnWindowXamlHostSizeChanged;
+
+            // Windows.UI.Xaml.Application object is required for loading custom control metadata.  If a custom
+            // Application object is not provided by the application, the host control will create one (XamlApplication).
+            // Instantiation of the application object must occur before creating the DesktopWindowXamlSource instance.
+            // If no Application object is created before DesktopWindowXamlSource is created, DestkopWindowXamlSource
+            // will create a generic Application object unable to load custom UWP XAML metadata.
+            Microsoft.Toolkit.Win32.UI.XamlHost.XamlApplication.GetOrCreateXamlApplicationInstance(ref _application);
+
+            // Create an instance of the WindowsXamlManager. This initializes and holds a
+            // reference on the UWP XAML DXamlCore and must be explicitly created before
+            // any UWP XAML types are programmatically created.  If WindowsXamlManager has
+            // not been created before creating DesktopWindowXamlSource, DesktopWindowXaml source
+            // will create an instance of WindowsXamlManager internally.  (Creation is explicit
+            // here to illustrate how to initialize UWP XAML before initializing the DesktopWindowXamlSource.)
+            _windowsXamlManager = Windows.UI.Xaml.Hosting.WindowsXamlManager.InitializeForCurrentThread();
+
+            // Create DesktopWindowXamlSource, host for UWP XAML content
+            _xamlSource = new Windows.UI.Xaml.Hosting.DesktopWindowXamlSource();
+
+            _xamlSource.TakeFocusRequested += this.OnTakeFocusRequested;
+        }
+
+        /// <summary>
+        ///    Gets or sets XAML content for XamlContentHost
+        /// </summary>
+        [Browsable(false)]
+        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
+        public Windows.UI.Xaml.UIElement ChildInternal
```

Comment:
I wrote code in other branches to keep from exposing UIElement to clients of this class.  Do we need that?  Should I push that solution here?

Reply:
I don't think we should expose anything with the name `Internal`. Can you propose a change to this branch with the changes for review?